src/rpdk/core/contract/resource_client.py

```diff
@@ -41,6 +41,50 @@ def prune_properties(document, paths):
     return document
 
 
+def prune_null_properties_in_path(model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    document = {"properties": model.copy()}
+    for path in paths:
+        try:
+            prop, resolved_path, parent = traverse(document, path)
+        except LookupError:
+            pass  # not found means nothing to delete
+        else:
+            if not prop:
+                key = resolved_path[-1]
+                del parent[key]
+    return document["properties"]
+
+
+def prune_properties_from_model_in_path(output_model, input_model, paths):
+    """Prune given properties from a document.
+
+    This assumes properties will always have an object (dict) as a parent.
+    The function modifies the document in-place, but also returns the document
+    for convenience. (The return value may be ignored.)
+    """
+    output_document = {"properties": output_model.copy()}
+    input_document = {"properties": input_model.copy()}
+    LOG.debug("This is output %s, input %s", output_document, input_document)
+    for path in paths:
+        try:
+            LOG.debug("This is path %s", path)
+            _prop, _resolved_path, _parent = traverse(input_document, path)
+        except LookupError:
+            LOG.debug("in exception")
+            _prop, resolved_path, parent = traverse(output_document, path)
+            LOG.debug("Not exception")
+            key = resolved_path[-1]
+            del parent[key]
```

Comment:
I still don't follow the logic. Why we need to `traverse` the output in the LookupError, and it only delete the output_document path?

Reply:
This method is used to remove createOnly properties from the output. For example we call create handler with input (the input does not contain createOnly properties). The create handler gives us an output model (the create handler generated the createOnly property) Now, when we compare ```created_input == create_output``` the two models are going to be different. To avoid this issue, ```prune_null_properties_in_path``` prunes all createOnly properties from the input_model which are null/empty. Then, the ```prune_properties_from_model_in_path``` method removes all the createOnly properties from the output_model if they don't exist in the input_model (Lookup exception is encountered) This is done because the input_model would have been pruned in the previous method and because there can be multiple createOnly properties we will need to iterated through the list.